gluetool/utils.py

```diff
@@ -1540,7 +1541,10 @@ def _timeout():
     while timeout is None or time.time() < end_time:
         logger.debug("calling callback function")
         ret = check()
-        if ret:
+
+        # If ret is an Tuple, use the first item to evaluate. This is useful if the wait function
+        # returns some output you want to process later ...
+        if isinstance(ret, tuple) and ret[0] or ret:
```

Comment:
thanks, bad day :)
I changed it to:
```
if (isinstance(ret, tuple) and ret[0]) or ret:
    logger.debug('check passed, assuming success')
    return ret
```
which must be also correct, due to short-circuit evaluation

Reply:
Sorry, still not good enough:

```python
>>> ret = (False, 'some dummy value')
>>>
>>> 'check passed!' if (isinstance(ret, tuple) and ret[0]) or ret else 'check failed!'
'check passed!'
>>>
```

The left side of `or` evaluates to `False` (because it is tuple & the first item is not true-ish), therefore no short-circuit happens - interpreter needs to evaluate the right side of `or` as well, and that's true because `ret` is not an empty tuple.

Reply:
ah right :) ok, will use your example :)